src/app/archive/archive_lib/processor.ml

```diff
@@ -82,11 +94,10 @@ module Token = struct
   let find_opt (module Conn : CONNECTION) =
     make_finder Conn.find_opt Caqti_request.find_opt
 
-  let add_if_doesn't_exist (module Conn : CONNECTION)
-      ~(owner : Account_id.t option) token_id =
+  let add_if_doesn't_exist (module Conn : CONNECTION) token_id =
     let open Deferred.Result.Let_syntax in
     let value = Token_id.(to_string token_id) in
-    match owner with
+    match Token_owners.find_owner token_id with
```

Comment:
Not finding an owner doesn't mean this is default token, right? (See the comment in `Token_owners` module definition)

Reply:
It might be safer to have the table contain options, so that a lookup should never fail.